src/Implement/MainManager.cpp

```diff
@@ -0,0 +1,107 @@
+#include "MainManager.h"
+
+
+namespace{
+
+    MainManager* onlyMainManager = nullptr;
+}
+
+void SigHandle(int sig, siginfo_t *info, void *ctx){
+
+    if(sig == SIGUSR1)
+    {
+        mainManager.RemoveByValue(info->si_pid);
+    }
+}
+
+void MainManager::InitInstance(const uint16_t &port_num){
+
+    this.port_num = port_num;
+    HttpListener listen_socket = HttpListener::GetInstance();
+    listen_socket.InitInstance(this.NewRequest, port_num);
+    listen_socket.StartListen();
+}
+
+MainManager* MainManager::GetInstance(){
+
+    if(onlyMainManager == nullptr){
+
+        return new MainManager();
```

Comment:
Is this class a singleton? if so, **onlyMainManager** might need an update when this instance is created.

Reply:
Done

Reply:
This is a reasonable review.